tuf/client/updater.py

```diff
@@ -1448,6 +1486,14 @@ def _verify_uncompressed_metadata_file(self, metadata_file_object,
     if not valid:
       raise securesystemslib.exceptions.BadSignatureError(metadata_role)
 
+    # For root metadata, verify the downloaded root metadata object with the
+    # new threshold of new signatures contained within the downloaded root
+    # metadata object
+    if valid and metadata_role == 'root':
+      valid = self._verify_root_sigs(metadata_signable)
```

Comment:
I also wouldn't call this function inside _verify_metadata_file(), because there is too much magic. Best to call it explicitly in the _update_root_metadata() code IMHO.

Reply:
Agreed, this does make `_verify_metadata_file()` much more magical. I opted to call it inside `_verify_metadata_file()` because this enable us to perform the check before the tempfile is persisted, and makes it easier to perform the check per download mirror.

If we do the check in `_update_root_metadata()` we need to clean up a file from the clients metadata directory and, perhaps less importantly, can only perform the check on a single file returned by `_update_root_metadata()`

Reply:
+1. I think the point of never writing unverified files is _critical_ and should probably be encoded in comments or docs... If we do not adhere to that it's almost impossible to ensure that our state of metadata stays valid.

Reply:
Also, we want to make sure that other functions that depend on this function do the same thing. Let's leave the magic alone for now, but document clearly why it is there, and not somewhere else.